native/src/gpxRouteApproximation.h

```diff
@@ -0,0 +1,55 @@
+#ifndef _OSMAND_GPX_ROUTE_APPROXIMATION_H
+#define _OSMAND_GPX_ROUTE_APPROXIMATION_H
+
+#include "CommonCollections.h"
+#include "commonOsmAndCore.h"
+
+struct RoutingIndex;
+struct RouteSegmentResult;
+struct GpxPoint;
+struct RoutingContext;
+class RoutePlannerFrontEnd;
+
+struct GpxRouteApproximation {
+	RoutingContext* ctx = nullptr;
+	RoutePlannerFrontEnd* router = nullptr;
+	int routeCalculations = 0;
+	int routePointsSearched = 0;
+	int routeDistCalculations = 0;
+	vector<SHARED_PTR<GpxPoint>> finalPoints;
+	vector<SHARED_PTR<RouteSegmentResult>> fullRoute;
+	int routeDistance;
+	int routeGapDistance;
+	int routeDistanceUnmatched;
```

Comment:
If you set default value by once for any member of class/structure , then need to set for all. Otherwise you got unexpected waste in iOS.
Best practice is set default values in constructor.

Reply:
It is all old code with outdated variables.

In Java, they are partially used for final debug information which is not supported in C++.

routeDistance not used, removed
routeGapDistance not used, removed
routeDistanceUnmatched not used, removed